facebook/facebook-json.c

```diff
@@ -314,3 +314,34 @@ gboolean fb_json_str_chk(const json_value *json, const gchar *name,
     *val = jv->u.string.ptr;
     return TRUE;
 }
+
+/**
+ * Backslash-escapes a string to make it safe for json. The returned string
+ * should be freed with #g_free() when no longer needed.
+ *
+ * *param input The string to escape
+ *
+ * @return The resulting string, or NULL on error.
+ **/
+gchar *fb_json_escape_string(const gchar *input)
+{
+    GString *out;
+    guint    i;
+
+    g_return_val_if_fail(input != NULL, NULL);
+
+    /* let's overallocate a bit */
+    out = g_string_sized_new(strlen(input) * 2);
+
+    for (i = 0; input[i]; i++) {
+        if (input[i] > 0 && input[i] < 0x20) {
+            g_string_append_printf(out, "\\u%04x", input[i]);
+            continue;
+        } else if (input[i] == '"' || input[i] == '\\') {
```

Comment:
Encase each lower level boolean expression in parenthesis.  Ex: `if ((A == B) || (C == D))`